### ISSUE TITLE
Solving the focus outline style

### ISSUE THREAD

**Issue by carolinan** [WordPress/twentytwentyfive#6]
This issue is inherited from Twenty Twenty-Four: https://github.com/WordPress/twentytwentyfour/issues/744
To comply with [WCAG 2.4.13 focus appearance](https://www.w3.org/TR/WCAG22/#focus-appearance), the focus indicator needs to be at least 2px thick and have a contrast ratio of at least 3:1 between the same pixels in the focused and unfocused states. See [Understanding success criterion](https://www.w3.org/WAI/WCAG22/Understanding/focus-appearance).
The open issue also suggests that the outline needs to be solid.

The editor lacks options for adjusting the outline for all focusable elements, so this needs to be solved with custom CSS.
To clarify, it is not enough to only style the focus outline for links and buttons.
The options are to add the CSS to theme.json or enqueue a stylesheet.
In the discussions about how to solve this for Twenty Twenty-Four, it was brought up that if the style is added to theme.json, users can remove it from the CSS field in the Site Editor, which would make the theme less accessible (Unless the user also adds an improved style).

Related Gutenberg issue: https://github.com/WordPress/gutenberg/issues/38277

**Comment by richtabor** [comment]
> To clarify, it is not enough to only style the focus outline for links and buttons.

Can you clarify this?

**Comment by carolinan** [comment]
The focus style needs to be accessible on **all** focusable elements.
The WCAG does not limit the guideline to links and buttons, it applies to all focusable elements.

**Comment by joedolson** [comment]
E.g., `input`, `textarea`, `select`, `input`, `button`, `tabindex=!-1`, `contentEditable`, etc.

**Comment by carolinan** [comment]
When I was working on the back to top block, I found that it was not reasonable for _me_ to create a complete list of focusable elements, I ended up with: 
https://github.com/WordPress/gutenberg/pull/50600/files#diff-7d8a7770b0e394828f51ea87f4de55be62d6e2f91a3a3951380ae8ebc4f15327R16

That is why in TT4, the style right now has the selector `where(.wp-site-blocks *:focus)`

This causes some styles for the navigation block that users have reported are unwanted, and is still discussed in
https://core.trac.wordpress.org/ticket/60808#comment:44

**Comment by germanfrelo** [comment]
This is my suggestion:

```css
/**
 * Better focus indicators.
 * Source: https://www.sarasoueidan.com/blog/focus-indicators/#a-%E2%80%98universal%E2%80%99-focus-indicator
 */

:where(.wp-site-blocks *:focus-visible) {
	outline: 0.375rem double #000;
	box-shadow: 0 0 0 0.25rem #fff;
}
```

**Comment by jasmussen** [comment]
Would it be an option to rely on browser default focus styles, and not specify styled ones in the theme itself? In most browsers, this provides very clear focus styles that work regardless of section or background color. 

There is [a new proposal for how to style pseudo states](https://github.com/WordPress/gutenberg/issues/38277#issuecomment-2306449475), including focus, in the block editor, which is likely easier to implement on the dev-side. It might still not make it for 6.7, and thus be leveraged for this. But by not providing custom CSS for focus styles, relying on the browser, it would keep the options open for users customizing the theme, once those options finally land.

**Comment by carolinan** [comment]
No, not according to [WCAG 2.4.13 focus appearance](https://www.w3.org/TR/WCAG22/#focus-appearance), and the reporter.

**Comment by carolinan** [comment]
Limiting any accessibility features to core blocks only is also a problem.

**Comment by jasmussen** [comment]
I guess you're referring to this part under _exceptions_:

> The focus indicator is determined by the [user agent](https://www.w3.org/TR/WCAG22/#dfn-user-agents) and cannot be adjusted by the author

Since we _can_ adjust the focus indicator, we should. Is that the gist? 

> Limiting any accessibility features to core blocks only is also a problem.

You should be able to style base elements in the block editor.

**Comment by carolinan** [comment]
This is already addressed above: https://github.com/WordPress/twentytwentyfive/issues/6#issuecomment-2259497879

**Comment by jasmussen** [comment]
I noticed, and that's fine. It did cause [issues in the editor for TT4](https://core.trac.wordpress.org/ticket/60535), and we can be mindful of those. The other motivation is simply that some browsers provide better focus styles than we can, using custom CSS. Thank you for answering my question. If it's addressed, shall we close this one?

**Comment by carolinan** [comment]
The outline style is only enqueued on the front. Because I found that it interfered with the style of blocks when they were selected in the editor.
